src/System.Runtime/tests/System/StringTests.netcoreapp.cs

```diff
@@ -12,6 +12,34 @@ namespace System.Tests
     public partial class StringTests
     {
         [Theory]
+        [InlineData(0, 0)]
+        [InlineData(3, 1)]
+        public static void Ctor_CharSpan_EmptyString(int length, int offset)
+        {
+            Assert.Same(string.Empty, new string(new ReadOnlySpan<char>(new char[length], offset, 0)));
+        }
+
+        [Theory]
+        [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0' }, 0, 8, "abcdefgh")]
+        [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0' }, 0, 9, "abcdefgh\0")]
+        [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0', 'i', 'j', 'k' }, 0, 12, "abcdefgh\0ijk")]
+        [InlineData(new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', '\0' }, 2, 3, "cde")]
+        [InlineData(new char[] { '\0' }, 0, 1, "\0")]
+        [InlineData(new char[] { 'a', 'b', 'c' }, 0, 0, "")]
+        [InlineData(new char[] { 'a', 'b', 'c' }, 1, 0, "")]
+        [InlineData(new char[] { 'a', 'b', 'c' }, 0, 1, "a")]
+        [InlineData(new char[] { 'a', 'b', 'c' }, 1, 1, "b")]
```

Comment:
Nit: This feels like way too many redundant test cases, not exercising any unique paths. I think this list can be shorter.